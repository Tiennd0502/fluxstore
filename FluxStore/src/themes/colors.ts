export const colors = {
  transparent: 'transparent',
  text: {
    50: '#33302eb3',
    100: '#a3a5ad',
    200: '#cbcdd8',
    300: '#8a8a8f',
    400: '#33302e',
    500: '#1d1f22',
    600: '#9b9b9b',
    700: '#777e90',
    800: '#000',
    900: '#1d1f22',
  },
  background: {
    50: '#00000080',
    100: '#f3f3f3',
    200: '#23262f',
    300: '#E7E8E9',
    400: '#464447',
    500: '#fff',
    600: '#3A2C27',
    700: '#fcfcfd',
    800: '#9d9d9d',
    900: '#1d1f22',
  },
  border: {
    100: '#d6d6d6',
    200: '#b1b5c3',
    300: '#332218',
    400: '#3A2C27',
  },
  error: {
    400: '#f08c7d',
    500: '#981b1f',
  },
  success: {
    300: '#508a7b',
    500: '#5ece7b',
  },
  linearGradient: {
    500: ['#353945', '#777e90'],
  },
};
