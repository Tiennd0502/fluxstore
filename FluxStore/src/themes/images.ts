export const Images = {
  welcome: require('@/assets/images/welcome.png'),
};
