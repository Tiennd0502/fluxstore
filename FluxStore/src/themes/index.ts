export * from './colors';
export * from './typography';
export * from './components';
export * from './metrics';
export * from './images';
export * from './theme';
