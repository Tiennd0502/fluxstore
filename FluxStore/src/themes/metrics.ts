export const borderRadius = {
  xs: 8,
  sm: 10,
  md: 15,
  lg: 20,
  xl: 30,
  circle: '50%',
};
