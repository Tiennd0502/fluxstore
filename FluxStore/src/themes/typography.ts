export const fontWeights = {
  regular: '400',
  medium: '500',
  semiBold: '600',
  bold: '700',
} as const;

export const fontSize = {
  xxl: 26,
  xl: 25,
  lg: 24,
  md: 20,
  base: 18,
  sm: 16,
  xs: 14,
  xxs: 13,
  tiny: 12,
  micro: 11,
  mini: 10,
};
