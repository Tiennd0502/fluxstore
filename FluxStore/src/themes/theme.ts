import {Theme} from '@/interfaces';
import {colors} from './colors';

export const lightThemes: Theme = {
  transparent: 'transparent',
  primary: colors.black[500],
  secondary: colors.white[700],
  tertiary: colors.black[300],
  error: colors.red[500],
  success: colors.green[500],
  link: colors.green[200],
  text: {
    default: colors.black[500],
    light: colors.white[500],
    primary: colors.black[600],
    secondary: colors.gray[800],
    tertiary: colors.gray[700],
    quaternary: colors.gray.A300,
    quinary: colors.black[50],
    senary: colors.gray[500],
    link: colors.green[200],
    error: colors.red[500],
    success: colors.green[500],
  },
  background: {
    default: colors.white[500],
    primary: colors.black[300],
    secondary: colors.gray[600],
    tertiary: colors.white.A900,
    quaternary: colors.white[400],
    input: colors.white[800],
  },
  border: {
    default: colors.black.A50,
    primary: colors.gray[50],
    secondary: colors.gray.A50,
    tertiary: colors.white[500],
  },
};

export const darkThemes: Theme = {
  transparent: 'transparent',
  primary: colors.white[500],
  secondary: colors.black[800],
  tertiary: colors.white[600],
  error: colors.red[500],
  success: colors.green[500],
  link: colors.green[200],
  text: {
    default: colors.white[500],
    light: colors.black[500],
    primary: colors.white[600],
    secondary: colors.gray.A100,
    tertiary: colors.gray[200],
    quaternary: colors.gray[200],
    quinary: colors.gray.A200,
    senary: colors.gray.A200,
    link: colors.green[200],
    error: colors.red[500],
    success: colors.green[500],
  },
  background: {
    default: colors.black.A100,
    primary: colors.white[600],
    secondary: colors.black[800],
    tertiary: colors.black[800],
    quaternary: colors.black[800],
    input: colors.gray.A400,
  },
  border: {
    default: colors.white[600],
    primary: colors.gray[200],
    secondary: colors.gray[800],
    tertiary: colors.white[500],
  },
};
