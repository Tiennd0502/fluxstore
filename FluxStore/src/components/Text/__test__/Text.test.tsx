import {render, screen} from '@testing-library/react-native';

import Text from '..';
import {colors, fontSizes} from '@/themes';

jest.mock('@/hooks', () => ({
  useThemeStore: jest.fn(() => ({
    theme: {
      fonts: {
        primary: {
          bold: 'Primary-Bold',
          regular: 'Primary-Regular',
        },
        secondary: {medium: 'Secondary-Medium'},
        tertiary: {regular: 'Tertiary-Regular'},
        default: {
          bold: 'Default-Bold',
          medium: 'Default-Medium',
          regular: 'Default-Regular',
        },
      },
      text: {
        default: '#000000',
        primary: '#111111',
        quaternary: '#222222',
      },
    },
  })),
}));

describe('Text', () => {
  const props = {
    children: 'Lorem ipsum dolor sit amet',
  };
  it('renders default', () => {
    const {toJSON} = render(<Text {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with variant heading', () => {
    const {toJSON} = render(<Text {...props} variant="heading" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with variant title', () => {
    const {toJSON} = render(<Text {...props} variant="title" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with variant subTitle', () => {
    const {toJSON} = render(<Text {...props} variant="subTitle" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with variant description', () => {
    const {toJSON} = render(<Text {...props} variant="description" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with custom color, fontsize', () => {
    const {toJSON} = render(
      <Text {...props} color={colors.green[500]} fontSize={fontSizes['3xl']} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('falls back to default font if fonts.primary.bold is missing', () => {
    (useThemeStore as jest.Mock).mockReturnValueOnce({
      theme: {
        fonts: {
          default: {bold: 'Default-Bold'},
        },
        text: {
          default: '#000000',
        },
      },
    });

    render(<Text variant="heading">Fallback Font</Text>);
    const element = screen.getByText('Fallback Font');
    console.log('STYLE:', element.props.style); // helpful debug

    const fontFamilyStyle = (element.props.style || []).find((s: any) => s && s.fontFamily);

    expect(fontFamilyStyle).toBeDefined();
  });
});
