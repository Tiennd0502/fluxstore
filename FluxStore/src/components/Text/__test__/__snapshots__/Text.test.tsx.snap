// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders default 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#737680",
        "fontFamily": "ProductSansRegular",
        "fontSize": 14,
      },
      {},
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text renders with custom color, fontsize 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#737680",
        "fontFamily": "ProductSansRegular",
        "fontSize": 14,
      },
      {
        "color": "#5ece7b",
        "fontSize": 40,
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text renders with variant description 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#1d1f22",
        "fontFamily": "ProductSansRegular",
        "fontSize": 12,
      },
      {},
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text renders with variant heading 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#000",
        "fontFamily": "ProductSansBold",
        "fontSize": 24,
        "fontWeight": 700,
      },
      {},
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text renders with variant subTitle 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#000",
        "fontFamily": "RobotoRegular",
        "fontSize": 16,
        "fontWeight": 700,
      },
      {},
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text renders with variant title 1`] = `
<Text
  style={
    [
      {
        "letterSpacing": 0,
        "textAlign": "left",
        "textAlignVertical": "center",
      },
      {
        "color": "#000",
        "fontFamily": "NunitoSemiBold",
        "fontSize": 20,
        "fontWeight": 700,
      },
      {},
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;
